### DISCUSSION TITLE
Need 2.0 release

### DISCUSSION THREAD

**Discussion by debragail** [MichalLytek/type-graphql#1544] [Q&A]
1.1.1 is from 2020 and we need the 2.0 release.  What can we do to get a stable release?

**Comment by MichalLytek** [comment]
Try using the beta.3 version and report feedback if something is broken/missing.

**Reply by debragail** [reply]
Cool, but why is it beta?

**Reply by MichalLytek** [reply]
As it does not have all the stuff for full 2.0 release yet and might have some regressions.

**Reply by debragail** [reply]
@MichalLytek I have several clients using this package in production and I'm going to have to fork this and maintain this myself.  Just trying to understand the roadmap so I can choose how to move forward.  Thanks.

**Reply by MichalLytek** [reply]
The more feedback I receive from 2.0 beta, the faster it will be considered stable and released.

**Comment by andreseloysv** [comment]
> Try using the beta.3 version and report feedback if something is broken/missing.

What if we release version 2.0 and gather feedback? If any issues arise, they can be addressed more swiftly in a production environment, or we can revert to a previous version. Everyone is eagerly anticipating the 2.0 release. Relying on betas is not feasible, especially since version 1 has security issues. I’m a fan of this library, but I’m concerned that people might turn to other alternatives, which wouldn’t be good for the project. It might be more effective to communicate a specific goal like, ‘I need to achieve X before the release, how can the community assist?

**Reply by MichalLytek** [reply]
The v2.0 release contains tons of breaking changes. And I need feedback about them, so we can change the API without releasing 3.0 and 4.0.

Betas are tested and stable, just there might be new breaking changes in new beta releases. Shall I name them RC, so it's safer to use by corporates?

**Reply by andreseloysv** [reply]
RCs might not be the best route. The most viable option appears to be releasing a stable version. There’s no issue with having a major release every six months. That approach is preferable to using betas or RCs for years.

**Reply by vjgene** [reply]
If it is considered stable enough, a RC release is a better option. I have been using this library and it appears stable, however the beta tag concerns the team when it's a runtime dependency. So, at least if there is no release, I would request a feature that generates graphql compatible resolvers instead of using resolvers/create.ts code which relies on IOC container and convertArgs function at runtime. This way I can use type-graphql as only devDependency.
